### DISCUSSION TITLE
Want to run the code in debuging mode by puting breakpoints

### DISCUSSION THREAD

**Discussion by abdullah2146** [idaholab/moose#21337] [Q&A General] [answered]
Hi,

I want to know is there any way that I can put a breakpoint at any code line and it will stop at that point to see the values. 

I'm also trying to use another way of std::cout<<var_name<<std::endl; , but the problem is I need to compile the code again and it takes time. So is there any other way to skip that time?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Yes you definitely can do.
See the manual for gdb
https://ftp.gnu.org/old-gnu/Manuals/gdb/html_node/gdb_27.html

We are also building up documentation for working with visual studio and use the connection to the debugger there. It's not ready yet

Guillaume

**Reply by aaelmeli** [reply]
> Hello
> 
> Yes you definitely can do. See the manual for gdb https://ftp.gnu.org/old-gnu/Manuals/gdb/html_node/gdb_27.html
> 
> We are also building up documentation for working with visual studio and use the connection to the debugger there. It's not ready yet
> 
> Guillaume

Hi @GiudGiud 
Have you had any documentation for the vs code?

**Reply by GiudGiud** [reply]
Best documentation is there

https://mooseframework.inl.gov/help/development/VSCode.html

Still nothing on debugging within VScode though.

**Reply by adhithya96** [reply]
Any updates on this ? It's hard to navigate through the moose code to understand the workflow. Setting some breakpoints in an IDE will be a lifesaver.

**Reply by GiudGiud** [reply]
@dschwen do you know anyone who could write this down
